dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/database/migration/V9_02_02__DML_Fill_urlaliases.java

```diff
@@ -0,0 +1,290 @@
+import java.util.UUID;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.flywaydb.core.internal.jdbc.JdbcTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
+
+/**
+ * Rollback:
+ *
+ * <ul>
+ *   <li>drop table url_aliases;
+ *   <li>delete from flyway_schema_history where version='9.02.02';
+ * </ul>
+ */
+@SuppressWarnings("checkstyle:typename")
+public class V9_02_02__DML_Fill_urlaliases extends BaseJavaMigration {
+
+  // Cannot be more elegant, since we want to allow null values
+  private static final Map<EntityType, String> ENTITYMIGRATIONTABLES = new HashMap<>();
+
+  static {
+    ENTITYMIGRATIONTABLES.put(AGENT, null);
+    ENTITYMIGRATIONTABLES.put(ARTICLE, "articles");
+    ENTITYMIGRATIONTABLES.put(AUDIO, null);
+    ENTITYMIGRATIONTABLES.put(BOOK, null);
+    ENTITYMIGRATIONTABLES.put(COLLECTION, "collections");
+    ENTITYMIGRATIONTABLES.put(CORPORATE_BODY, "corporatebodies");
+    // FIXME: This must be removed dynamically by the configuration
+    // ENTITYMIGRATIONTABLES.put(DIGITAL_OBJECT, "digitalobjects");
+    ENTITYMIGRATIONTABLES.put(ENTITY, null);
+    ENTITYMIGRATIONTABLES.put(EVENT, null);
+    ENTITYMIGRATIONTABLES.put(EXPRESSION, null);
+    ENTITYMIGRATIONTABLES.put(FAMILY, "familynames");
+    ENTITYMIGRATIONTABLES.put(GEOLOCATION, "geolocations");
+    ENTITYMIGRATIONTABLES.put(HEADWORD_ENTRY, null);
+    ENTITYMIGRATIONTABLES.put(IMAGE, null);
+    ENTITYMIGRATIONTABLES.put(ITEM, "items");
+    ENTITYMIGRATIONTABLES.put(MANIFESTATION, null);
+    ENTITYMIGRATIONTABLES.put(OBJECT_3D, null);
+    ENTITYMIGRATIONTABLES.put(PERSON, "persons");
+    ENTITYMIGRATIONTABLES.put(PLACE, null);
+    ENTITYMIGRATIONTABLES.put(PROJECT, "projects");
+    ENTITYMIGRATIONTABLES.put(TOPIC, "topics");
+    ENTITYMIGRATIONTABLES.put(VIDEO, null);
+    ENTITYMIGRATIONTABLES.put(WEBSITE, "websites");
+    ENTITYMIGRATIONTABLES.put(WORK, "works");
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V9_02_02__DML_Fill_urlaliases.class);
+
+  private final SlugGenerator slugGenerator = new SlugGenerator();
+
+  @Override
+  public void migrate(Context context) throws Exception {
+
+    // TODO Retrieve configuration and filter ENTITIYMIGRATIONTABLES accordingly!  (and remove
+    // comment accordingly)
+
+    final SingleConnectionDataSource connectionDataSource =
+        new SingleConnectionDataSource(context.getConnection(), true);
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(connectionDataSource.getConnection());
+
+    // Webpages have to be migrated individually, since their URLAliases depend on the
+    // the websites, they are bound, too
+    migrateWebpages(jdbcTemplate);
+
+    // All other entities have only generic URLAliases
+    ENTITYMIGRATIONTABLES.forEach(
+        (entityType, tableName) -> {
+          if (tableName != null) {
```

Comment:
Wozu gibts dann oben Schlüssel mit `null` als Wert, wenn für die sowieso nichts gemacht wird?

Reply:
Die Idee war, in der Map eine vollständige Abbildung aller Entity-Typen (`de.digitalcollections.model.identifiable.entity.EntityType.*`) zu hinterlegen und diejenigen, für die es (noch) keine eigene Tabelle in der Datenbank gibt, `null` zu hinterlegen.